stackslib/src/net/api/getaccount.rs

```diff
@@ -146,76 +146,80 @@ impl RPCRequestHandler for RPCGetAccountRequestHandler {
 
         let account_opt_res =
             node.with_node_state(|_network, sortdb, chainstate, _mempool, _rpc_args| {
-                chainstate.maybe_read_only_clarity_tx(&sortdb.index_conn(), &tip, |clarity_tx| {
-                    clarity_tx.with_clarity_db_readonly(|clarity_db| {
-                        let key = ClarityDatabase::make_key_for_account_balance(&account);
-                        let burn_block_height =
-                            clarity_db.get_current_burnchain_block_height().ok()? as u64;
-                        let v1_unlock_height = clarity_db.get_v1_unlock_height();
-                        let v2_unlock_height = clarity_db.get_v2_unlock_height().ok()?;
-                        let v3_unlock_height = clarity_db.get_v3_unlock_height().ok()?;
-                        let (balance, balance_proof) = if with_proof {
-                            clarity_db
-                                .get_data_with_proof::<STXBalance>(&key)
-                                .ok()
-                                .flatten()
-                                .map(|(a, b)| (a, Some(format!("0x{}", to_hex(&b)))))
-                                .unwrap_or_else(|| (STXBalance::zero(), Some("".into())))
-                        } else {
-                            clarity_db
-                                .get_data::<STXBalance>(&key)
-                                .ok()
-                                .flatten()
-                                .map(|a| (a, None))
-                                .unwrap_or_else(|| (STXBalance::zero(), None))
-                        };
+                chainstate.maybe_read_only_clarity_tx(
+                    &sortdb.index_handle_at_tip(),
```

Comment:
Please use something fallible here

Reply:
This gets used all over the place in this PR, so I'm just going to summarize my feelings in a review comment instead.